aten/src/TH/THRandom.cpp

```diff
@@ -10,14 +13,20 @@
 #define n _MERSENNE_STATE_N
 #define m _MERSENNE_STATE_M
 
+struct mutex_s {
+  std::shared_ptr<std::mutex> mutex;
```

Comment:
Why a shared ptr? It should work fine without the extra indirection. It’s not shared